sympy/printing/tests/test_fcode.py

```diff
@@ -685,3 +686,29 @@ def test_fcode_For():
     assert sol == ("      do x = 0, 10, 2\n"
                    "         y = x*y\n"
                    "      end do")
+
+def test_fcode_Declaration():
+    def check(expr, ref, **kwargs):
+        assert fcode(expr, standard=95, source_format='free', **kwargs) == ref
+
+    i = symbols('i', integer=True)
+    var1 = Variable.deduced(i)
+    dcl1 = Declaration(var1)
+    check(dcl1, "integer :: i")
+
+
+    x, y = symbols('x y')
+    var2 = Variable(x, {value_const}, float32)
+    dcl2b = Declaration(var2, 42)
+    check(dcl2b, 'real(4), parameter :: x = 42')
+
+    var3 = Variable(y, (), Type('bool'))
+    dcl3 = Declaration(var3)
+    check(dcl3, 'logical :: y')
+
+    check(float32, "real(4)")
+    check(float64, "real(8)")
+    check(Type('real'), "real(4)", precision=float32)
+    check(Type('real'), "real(4)", precision=6)
+    check(Type('real'), "real(8)", precision=float64)
+    check(Type('real'), "real(8)", precision=15)
```

Comment:
Actually it's usually just `prec` for binary precision. We only used the longer name `precision` in Float because we have to change the behavior (previously `prec` was decimal precision).  (side note, I see no deprecation issue was opened for this)